hierarchicalforecast/methods.py

```diff
@@ -461,8 +462,33 @@ def min_trace(S: np.ndarray,
     if any(eigenvalues < 1e-8):
         raise Exception(f'min_trace ({method}) needs covariance matrix to be positive definite.')
     
-    R = S.T @ np.linalg.pinv(W)
-    P = np.linalg.pinv(R @ S) @ R
+    W_inv = np.linalg.pinv(W)
+    if nonnegative:
+        if bootstrap:
+            raise Exception('nonnegative reconciliation is not compaitble with bootstrap forecasts')
+        if idx_bottom is None:
+            raise Exception('idx_bottom needed for nonnegative reconciliation')
+        # compute P for nonnegative reconciliation
+        warnings.warn('Replacing negative forecasts with zero.')
```

Comment:
I suppose we are using a simplified version of NN reconciliation, deviating from its [original R implementation](https://github.com/earowang/hts/blob/master/R/bpv.R).

If that is the case can we comment the line of thought and differences versus R's version in this section?